src/helpers/query/administrations.js

```diff
@@ -145,6 +145,21 @@ export const administrationPageFetcher = async (
 
   let administrations = await mapAdministrations(administrationData);
 
+  if (siteId) {
+    orgs = await fetchOrgsBySite(siteId);
+    orgs.push({ id: siteId });
+
+    administrations = administrations.filter((administration) => {
+      return orgs.some(
+        (org) =>
+          administration.assignedOrgs.districts.includes(org.id) ||
+          administration.assignedOrgs.schools.includes(org.id) ||
+          administration.assignedOrgs.classes.includes(org.id) ||
+          administration.assignedOrgs.groups.includes(org.id),
+      );
+    });
+  }
+
```

Comment:
Rather than doing this on the client, we should update the function getAdministration function on the backend. We don't want to be doing any filtering here. The administration documents now have a `siteId` property which we can easily get all documents for the given site.

Reply:
Did you look in depth at `getAdministrations `? I believe I updated it to filter by siteId. It's possible there's just a bug.